gap/aut-basics.gi

```diff
@@ -605,46 +605,58 @@ end);
 ## Produces the disjoint union of the automata A and B
 ##
 InstallGlobalFunction(UnionAutomata, function(A, B)
-    local   QA,  T,  a,  i,  I,  F;
-
-    if not (IsAutomatonObj(A) and IsAutomatonObj(B)) then
-        Error("The arguments must be two automata");
-    fi;
-    if A!.type = "nondet" then
-        A := NFAtoDFA(A);
+    local  QA, T, a, i, I, F, mA,mB;
+    
+    if not (IsAutomatonObj( A ) and IsAutomatonObj( B ))  then
+    	Error( "The arguments must be two automata" );
     fi;
+    
     if A!.type = "epsilon" then
-        A := NFAtoDFA(EpsilonToNFA(A));
-    fi;
-    if B!.type = "nondet" then
-        B := NFAtoDFA(B);
+    	mA := AlphabetOfAutomaton(A)-1;
```

Comment:
This line is never exercised by the tests according to Codecov. Dunno how hard it would be to add such a test?